src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -0,0 +1,255 @@
+import { BusinessLookup } from '@bcrs-shared-components/business-lookup'
+import { BusinessIF, BusinessLookupIF, EmptyBusinessLookup } from '@/interfaces'
+
+@Component({
+  components: {
+    BusinessLookup
+  }
+})
+export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
+  @Getter(useStore) getAmalgamatingBusinesses!: Array<BusinessIF>
+  @Getter(useStore) isAmalgamationFilingHorizontal!: boolean
+  @Getter(useStore) isRoleStaff!: boolean
+
+  @Action(useStore) setAmalgamatingBusinesses!: (x: Array<BusinessIF>) => void
+
+  // Local properties
+  amalgamatingBusinessesValid = false
+  amalgamatingBuinesses = []
+  initialBusinessLookupObject = EmptyBusinessLookup
+
+  // Add an Amalgamating Business button properties
+  isAddingAmalgamatingBusiness = false
+  addAmalgatingBusinessDisabled = false
+
+  // Add an Amalgamating Foreign Business button properties
+  isAddingAmalgamatingForeignBusiness = false
+  addAmalgatingForeignBusinessDisabled = false
+
+  readonly BusinessLookupServices = BusinessLookupServices
+
+  // Cancel button in "Add an Amalgamating Business" is pressed.
+  addAmalgamatingBusinessCancel (): void {
+    this.isAddingAmalgamatingBusiness = false
+    // Enable buttons
+    this.addAmalgatingBusinessDisabled = false
+    this.addAmalgatingForeignBusinessDisabled = false
+  }
+
+  // "Add an Amalgamating Business" button is pressed.
+  onAddBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = true
+    this.isAddingAmalgamatingForeignBusiness = false
+    // Disable buttons
+    this.addAmalgatingBusinessDisabled = true
+    this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // "Add an Amalgamating Foreign Business" button is pressed.
+  onAddForeignBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = false
+    this.isAddingAmalgamatingForeignBusiness = true
+    // Disable buttons (Please comment out the lines below or remove in part 2)
+    // this.addAmalgatingBusinessDisabled = true
+    // this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // Add to the amalgamating businesses array.
+  // If EP (A type), cannot be part of short form horizontal amalgamation.
+  pushToAmalgamatingBusinesses (business: any): void {
+    if (this.isAmalgamationFilingHorizontal) {
+      if (business.legalType !== 'A') this.amalgamatingBuinesses.push(business)
+    } else {
+      this.amalgamatingBuinesses.push(business)
+    }
+  }
+
+  async setAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
+    // Get the amalgamating business information
+    // Will have a different format depending on the business
+    let business = await LegalServices.fetchBusinessInfo(businessLookup.identifier)
+      .then((response) => {
+        return response?.data?.business
+      }).catch(() => {
+        return businessLookup
+      })
+
+    // Get the address of the amalgamating business
+    if (businessLookup.identifier && business.foundingDate) {
```

Comment:
Why do you have these checks? I think you don't need them.

Reply:
The first one, you're right. For the second one, it is to handle whether the business was from LEAR. I'm unable to fetch the address if it's not a LEAR business. The foundingDate is a field only found in the big JSON above.

Reply:
> For the second one, it is to handle whether the business was from LEAR. I'm unable to fetch the address if it's not a LEAR business. The foundingDate is a field only found in the big JSON above.

I think what I proposed is a better way, but if we kept the current code then you should add a code comment saying why  you're checking the founding date. (I looked to see if you needed it for anything and it wasn't referenced, so I was confused.)